parsec/core/gui/invite_user_widget.py

```diff
@@ -1,206 +1,276 @@
-        if status == "cancelled":
-            self.registration_job = None
-            return
+    def greet(self, token):
+        GreetUserWidget.exec_modal(core=self.core, jobs_ctx=self.jobs_ctx, token=token, parent=self)
+        self.list_invitations()
 
-        if status == "registration-invite-bad-value":
-            errmsg = _("TEXT_INVITE_USER_BAD_USER_NAME")
-        elif status == "registration-invite-already-exists":
-            errmsg = _("TEXT_INVITE_USER_ALREADY_EXISTS")
-        elif status == "registration-invite-error":
-            errmsg = _("TEXT_INVITE_USER_WRONG_PARAMETERS")
-        elif status == "registration-invite-offline":
-            errmsg = _("TEXT_INVITE_USER_HOST_OFFLINE")
-        elif status == "registration-invite-timeout":
-            errmsg = _("TEXT_INVITE_USER_TIMEOUT")
-        else:
-            errmsg = _("TEXT_INVITE_USER_UNKNOWN_FAILURE")
-        show_error(self, errmsg, exception=self.registration_job.exc)
-        self.registration_job = None
-
-    def on_registration_success(self):
-        assert self.registration_job.is_finished()
-        assert self.registration_job.status == "ok"
-        show_info(self, _("TEXT_INVITE_USER_SUCCESS"))
-        self.registration_job = None
-        if self.dialog:
-            self.dialog.accept()
-        elif QApplication.activeModalWidget():
-            QApplication.activeModalWidget().accept()
-        else:
-            logger.warning("Cannot close dialog when inviting user")
+    def _on_cancel_invitation_success(self, job):
+        self.list_invitations()
 
-    def cancel_registration(self):
-        if self.registration_job:
-            self.registration_job.cancel_and_join()
+    def _on_cancel_invitation_error(self, job):
+        assert job.is_finished()
+        assert job.status != "ok"
+        show_error(self, _("TEXT_INVITE_USER_CANCEL_ERROR"), exception=job.exc)
 
-    def register_user(self):
-        if not self.line_edit_username.text():
-            show_error(self, _("TEXT_INVITE_USER_EMPTY_USER_NAME"))
-            return
+    def _on_invite_user_success(self, job):
+        self.list_invitations()
 
-        try:
-            new_user_id = UserID(self.line_edit_username.text())
-        except ValueError as exc:
-            show_error(self, _("TEXT_INVITE_USER_BAD_USER_NAME"), exception=exc)
-            return
+    def _on_invite_user_error(self, job):
+        assert job.is_finished()
+        assert job.status != "ok"
+        show_error(self, _("TEXT_INVITE_USER_INVITE_ERROR"), exception=job.exc)
 
-        token = core_generate_invitation_token()
-        try:
-            addr = BackendOrganizationClaimUserAddr.build(
-                self.core.device.organization_addr, user_id=new_user_id
-            )
-        except ValueError as exc:
-            show_error(self, _("TEXT_INVITE_USER_WRONG_PARAMETERS"), exception=exc)
-            return
+    def _on_list_invitations_success(self, job):
```

Comment:
I would be good to listen on `backend.invite.status_changed` event to know when to update the list
This typically useful to remove inivatiton deleted on other devices, or to switch the invitation status from idle to ready (and back again) when peer start claiming the invitation (it could be good to also send a gui notification in such case)

PS: `backend.invite.status_changed` doesn't exist yet, update should be made here first: 
https://github.com/Scille/parsec-cloud/blob/91116b4b277588b05634fc9fc54ae21a6b79aae3/parsec/core/backend_connection/authenticated.py#L75